### PR TITLE
mqtt: add additional tests for reason codes

### PR DESCRIPTION
**Pull request by satta**
This PR adds some more checks for situations related to https://forum.suricata.io/t/question-about-mqtt-detection/4890/ among others.

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/7323

### CODE REVIEW

**Comment by jufajardini** [comment]
Current Suricata PR: https://github.com/OISF/suricata/pull/11995

File: tests/mqtt-sub-rules/test.yaml [hunk 1]
```diff
@@ -109,3 +119,9 @@ checks:
       match:
         event_type: alert
         alert.signature: MQTT SUBSCRIBE topicY
+
+  - filter:
```
Comment (jufajardini) [inline tests/mqtt-sub-rules/test.yaml:123]: This one I think we'll have to add a `min-version` check for 8, before this can be merged.

**Review by jufajardini — CHANGES_REQUESTED** [review]
Thanks for this work :)

File: tests/mqtt-sub-rules/test.yaml (hunk 1 above)
Comment (satta) [reply tests/mqtt-sub-rules/test.yaml:123]: Oh, I didn't know you could also have `min-version` on individual filters. Sure, makes sense.

**Comment by satta** [comment]
Next PR: #2106

**Pull request closed without merging**